### PR TITLE
Closes #39 - remove unused vars

### LINKED ISSUE #39: Remove unused variables
While scanning the project with Flake8 (#33), we received several errors related to **unused imports**.

# Goal
Resolve all Flake8 errors related to unused imports.

# Task
The following Flake8 errors should be resolved:

- [x] ./api/read.py:156:9: F841 local variable 'user_categories' is assigned to but never used
- [x] ./api/read.py:187:9: F841 local variable 'problems' is assigned to but never used
- [x] ./api/write.py:17:1: F811 redefinition of unused 'require_post_params' from line 5                                                                                                                                                                                             
- [x] ./api/write.py:18:1: F811 redefinition of unused 'JsonResponse' from line 4                                                                                                                                                                                                    
- [x] ./api/write.py:19:1: F811 redefinition of unused 'login_required' from line 6
- [x] ./api/write.py:465:29: F841 local variable 'e' is assigned to but never used
- [x] ./api/models/civi.py:134:9: F841 local variable 'owner_id' is assigned to but never used

### PR DESCRIPTION
**Pull request by tractaylor**
I just commented out the code where the vars were defined except for the exception e variable in write.py, I just removed the e.

### CODE REVIEW

**Comment by brylie** [comment]
Add one of the [GitHub strings to associate this PR with its parent issue](https://help.github.com/articles/closing-issues-using-keywords/).

File: project/api/models/civi.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def score(self, request_acct_id=None):
         D = -1
         A = 1
         SA = 2
-        owner_id = self.author
+        #owner_id = self.author
```
Comment (brylie) [inline project/api/models/civi.py]: Don't leave commented code. Either it is used, or it should be removed.

**Comment by brylie** [comment]
Basically, the goal is to remove unused code. We shouldn't leave unused code, even in comments, unless there is a `# TODO` comment indicating why the code remains.

File: project/api/models/civi.py (hunk 1 above)
Comment (tractaylor) [reply project/api/models/civi.py]: ok owner_Id should be gone now.  Sorry about that.

**Comment by tractaylor** [comment]
after major confusion on my part, I think I finally have the owner_id removed.  https://github.com/CiviWiki/OpenCiviWiki/pull/71/commits/c0a761c1ef06579c8beeeb5a8447af46f0b4260b

I apologize for checking in changes into the wrong branch.

**Comment by tractaylor** [comment]
@brylie - about the "add one of the GitHub strings to associate this PR with its parent issue", can I do this after I created the PR?  Or should I have done this in one of the commit messages?  Sorry I am new to using the issues on github.

**Comment by brylie** [comment]
> about the "add one of the GitHub strings to associate this PR with its parent issue", can I do this after I created the PR?

You can add the string by editing the pull request description.

**Comment by mdwest** [comment]
Is this ready to be merged?

**Comment by tractaylor** [comment]
I think it was just waiting for another review after I made the changes requested by @brylie

**Comment by mdwest** [comment]
Great! @brylie can you confirm that and merge this pull request!

**Comment by brylie** [comment]
Remove any commented code that does not have a `TODO` comment.

**Comment by tractaylor** [comment]
@brylie did you see any comments that I missed?  From the last review, I had only missed one and I already made that change in https://github.com/CiviWiki/OpenCiviWiki/pull/71/commits/c0a761c1ef06579c8beeeb5a8447af46f0b4260b

Please let me know.

**Comment by brylie** [comment]
This PR looks good. Please remove any commented code you found, and we can merge.

**Comment by tractaylor** [comment]
This PR is for removing unused vars issue #39.  Issue #42 is in the other pull request.  I think I removed all the unused vars.

**Comment by brylie** [comment]
@tractaylor the problem is that this PR **adds** commented variables. Just remove the lines that I have noted above. 

In general, *every* PR should remove commented code (excepting code with TODO comments or that is intended as an example).

File: project/api/read.py [hunk 2]
```diff
@@ -1,16 +1,12 @@
 from django.http import JsonResponse, HttpResponseBadRequest
 from models import Account, Thread, Civi, Representative, Category, Activity
 #  Topic, Attachment, Category, Civi, Comment, Hashtag,
-from django.db.models import Q
 from django.contrib.auth.models import User
-from django.core.serializers.json import DjangoJSONEncoder
 from django.forms.models import model_to_dict
 from utils import json_response
-import json
 # from utils.custom_decorators import require_post_params
```
[the remark this answers is not in the document]
Comment (tractaylor) [reply project/api/read.py]: I have already removed line 10 in the other PR #72

File: project/api/read.py [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
 from legislation import sunlightapi as sun
 
-import tasks
 # # Create your views here.
 # @require_post_params(params=['id'])
```
[the remark this answers is not in the document]
Comment (tractaylor) [reply project/api/read.py]: I have already removed line 10 and 11 in the other PR #72

File: project/api/read.py [hunk 4]
```diff
@@ -151,9 +147,9 @@ def get_rep(request, rep_id):
 
 def get_feed(request):
     try:
-        a = Account.objects.get(user=request.user)
-        all_categories = Category.objects.values_list('id', flat=True)
-        user_categories = list(a.categories.values_list('id', flat=True)) or all_categories
+        # a = Account.objects.get(user=request.user)
```
[the remark this answers is not in the document]
Comment (tractaylor) [reply project/api/read.py]: I have already removed line line 150 in the other PR #72

File: project/api/read.py [hunk 5]
```diff
@@ -184,7 +180,7 @@ def get_thread(request, thread_id):
         c_data = [Civi.objects.serialize_s(ci) for ci in c]
         for idx, item in enumerate(c_data):
             problems[idx]['score'] = c_scores[idx]
-        problems = sorted(problems, key=lambda x: x['score'], reverse=True)
+        # problems = sorted(problems, key=lambda x: x['score'], reverse=True)
```
[the remark this answers is not in the document]
Comment (tractaylor) [reply project/api/read.py]: I have already removed line 183 in the other PR #72

**Comment by brylie** [comment]
OK, I double checked all the lines, and they have been removed in PR #72 as you mention. 

This PR looks great, and is nearly ready to merge! Please resolve conflicts, in the same way you did for PR #72.

**Comment by tractaylor** [comment]
@brylie I have resolved the conflicts.

**Comment by brylie** [comment]
Cool! You're 'old hat' at merging conflicts now ;-)

**Pull request merged**